server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts

```diff
@@ -23,9 +29,64 @@ ContactedPeopleRoute.get('/allContacts/:investigationId', (request: Request, res
 );
 
 ContactedPeopleRoute.post('/interactedContacts', (request: Request, response: Response) => {
-    graphqlRequest(SAVE_LIST_OF_CONTACTS, response.locals, { unSavedContacts: JSON.stringify(request.body)})
+    graphqlRequest(UPDATE_LIST_OF_CONTACTS, response.locals, { unSavedContacts: JSON.stringify(request.body)})
         .then((result: any) => response.send(result))
         .catch(error => response.status(500).json({error: 'failed to save all the contacts'}))
 });
 
+ContactedPeopleRoute.post('/excel', async (request: Request, response: Response) => {
+    const {contactEvent, contacts} = request.body;
+    const getIdFromResult = (result: any) => result?.nodes.length > 0 ? parseInt(result.nodes[0].id) : null;
+    const parsedContactsPromises = contacts.map(async (contactedPerson: InteractedContact) => {
+        const parsingVariables = {
+            city: contactedPerson.contactedPersonCity || '',
+            contactType: contactedPerson.contactType || '',
+            familyRelationship: contactedPerson.familyRelationship || '',
+        };
+
+        try {
+            const parsedForeignKeys = await graphqlRequest(GET_FOREIGN_KEYS_BY_NAMES, response.locals, parsingVariables)
+
+            const {allCities, allContactTypes, allFamilyRelationships} = parsedForeignKeys.data;
+            const contactedPersonCity = getIdFromResult(allCities),
+                contactType = getIdFromResult(allContactTypes),
+                familyRelationship = getIdFromResult(allFamilyRelationships);
+
+            return {
+                ...contactedPerson,
+                contactEvent,
+                contactedPersonCity,
+                contactType,
+                familyRelationship,
+            };
+        } catch (e) {
+            console.error(e);
+            return {
+                ...contactedPerson,
+                contactEvent,
+                contactedPersonCity: null,
+                contactType: null,
+                familyRelationship: null,
+            }
+        }
+
+    });
+
+    const parsedContacts = await Promise.all(parsedContactsPromises);
+    const mutationVariables = {
+        unSavedContacts: JSON.stringify(parsedContacts),
+    };
+
+    return graphqlRequest(UPDATE_LIST_OF_CONTACTS, response.locals, mutationVariables)
+        .then((result: any) => {
+            const hasErrors = result?.errors?.length > 0;
+            const status = hasErrors ? 500 : 200;
```

Comment:
export to a common function to use for future refactor

Reply:
sending the entire result object to a function just to save writing a condition seems pretty heavy to me?